src/com/facebook/buck/ide/intellij/IjProjectTemplateDataPreparer.java

```diff
@@ -503,6 +513,41 @@ private void addAndroidResourcePaths(
     }
   }
 
+  private void addAndroidCompilerOutputPath(
+      Map<String, Object> androidProperties,
+      IjModule module,
+      Path moduleBasePath) {
+    // This is the fallback default in case we can't find a valid build target name
+    String targetName = "src_release";
+
+    if (module.getTargets() != null) {
+      for (BuildTarget target : module.getTargets()) {
+        BuildRule buildRule = buildRuleResolver.getRule(target);
+        if ((buildRule instanceof AndroidLibrary
+            || buildRule instanceof JavaLibrary) && !target.getShortName().contains("aidl")) {
+          targetName = target.getShortName();
+          break;
+        }
+      }
+    }
+
+    Path rootModulePath = Paths.get(projectConfig.getProjectRoot());
+    Path moduleRelativePath = rootModulePath.relativize(moduleBasePath);
+
+    // IntelliJ may not be able to find classes on the compiler output path
+    // if the jar_spool_mode is set to direct_to_jar.
+    Path moduleLibClassesPath = Paths.get(
+        rootModulePath.toString(),
+        "buck-out/bin",
+        moduleRelativePath.toString(),
+        "lib__" + targetName + "__classes");
```

Comment:
Use this: https://github.com/facebook/buck/blob/44be031266fc118cb712802f1d67489291cde8d0/src/com/facebook/buck/jvm/java/DefaultJavaLibrary.java#L280-L282